virttest/utils_netperf.py

```diff
@@ -104,21 +104,39 @@ def __init__(self, address, netperf_path, md5sum="", netperf_source="",
             self.netperf_path = os.path.join(self.netperf_base_dir,
                                              self.netperf_file)
 
-        LOG.debug("Create remote session")
-        self.session = remote.remote_login(self.client, self.address,
-                                           self.port, self.username,
-                                           self.password, prompt,
-                                           linesep, timeout=360,
-                                           status_test_command=status_test_command)
+        self._session = None
+        self._session_attrs = (prompt, linesep, status_test_command)
+
+    @property
+    def session(self):
+        self._acquire_session()
+        return self._session
+
+    def _acquire_session(self):
+        if self._session is not None:
+            return
+        prompt, linesep, status_test_command = self._session_attrs
+        self._session = remote.remote_login(self.client, self.address,
+                                            self.port, self.username,
+                                            self.password, prompt,
+                                            linesep, timeout=360,
+                                            status_test_command=status_test_command)
+
+    def _release_session(self):
+        if self._session is None:
+            return
+        self._session.close()
+        self._session = None
 
     def env_cleanup(self, clean_all=True):
         clean_cmd = ""
         if self.netperf_dir:
             clean_cmd = "rm -rf %s" % self.netperf_dir
         if clean_all:
-            clean_cmd += " rm -rf %s" % os.path.join(self.remote_path,
```

Comment:
Nice catch.

Reply:
Actually the current change is still not correct, I'm working on a new version and will send it out soon, thanks.